### PR TITLE
WFNEWS-2419 route back to map page from full details

### PR DESCRIPTION
**Pull request by ssylver93**
Add coordinates and zoom level to query params when routing to the full details page in order to return the user to where they were just at on the map.
Also updating desktop version to route back to previous zoom level on the map for bans, evacs, area restrictions, rather than use a hardcoded value.
Includes some formatting updates applied automatically in VSCode.

### CODE REVIEW

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts [hunk 1]
```diff
@@ -48,8 +48,14 @@ document.getElementById('mobile-navigation-bar').style.display = 'block';
...
   back() {
     try {
-      if (this.params && this.params['source']) {
-        if (
+      if (this.params?.['source']) {
+        // use query params to determine the layer, coordinates and zoom level for routing back to the map
+        if ((this.params['source'] === 'map' || this.params['source']?.[0] === 'map')
+          && (this.params?.['type'] === 'area-restriction' || this.params?.['type'] === 'bans-prohibitions'
+            || this.params?.['type'].includes('evac'))
+        ) {
```
Comment (agordon-vivid) [inline client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: How would we summarize this logic into plain english?

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts [hunk 2]
```diff
@@ -75,18 +81,58 @@ document.getElementById('mobile-navigation-bar').style.display = 'block';
...
   }
 
-  async exit() {
-    this.router.navigate([ResourcesRoutes.DASHBOARD]);
+  exit() {
+    // use query params to determine the layer, coordinates and zoom level for routing back to the map
+    if ((this.params?.['source'] === 'map' || this.params?.['source']?.[0] === 'map')
+      && (this.params?.['type'] === 'area-restriction' || this.params?.['type'] === 'bans-prohibitions'
+        || this.params?.['type'].includes('evac'))
+    ) {
```
Comment (agordon-vivid) [inline client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: Same applies here

**Review by agordon-vivid — COMMENTED** [review]
Some of this logic seems a bit contrived.  Can we simplify or clarify it a bit better?

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts (hunk 1 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: If the user accessed the full details page from the map, and if the full details page contains one of area restrictions, fire bans, evac orders or evac alerts, use the backToMap function to route back to the map using the same layer and zoom level along with the appropriate coordinates for that type.
I updated the comments in the code with this explanation

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts (hunk 2 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: Logic updated to make it more readable

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts (hunk 1 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: Logic updated to make it more readable

**Review by yzlucas — APPROVED** [review]
Looks good, It did not mentioned in the ticket but I assume we want to do the same thing for Danger Rating? I think back button should take user to map with zoom level too. Can you confirm with Christine?

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts [hunk 3]
```diff
@@ -75,18 +80,78 @@ document.getElementById('mobile-navigation-bar').style.display = 'block';
...
     }
   }
 
-  async exit() {
-    this.router.navigate([ResourcesRoutes.DASHBOARD]);
+  exit() {
+    // If the user accessed the full details page from the map, and if the full details page contains one of area restrictions,
+    // fire bans, evac orders or evac alerts, use the backToMap function to route back to the map using the same layer and zoom level,
+    // along with the appropriate coordinates for that type
+    if (this.navigatedfromMapPage() && this.isCurrentlyNavigatedOn()) {
```
Comment (agordon-vivid) [inline client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]:
```suggestion
    if (this.navigatedfromMapPage() && this.isCurrentlyNavigatedOn(['area-restriction', 'bans-prohibitions', 'evac-orders', 'evac-alerts', 'danger-rating'])) {
```

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts [hunk 4]
```diff
@@ -75,18 +80,78 @@ document.getElementById('mobile-navigation-bar').style.display = 'block';
...
+    if (this.params?.['source'] === 'map' || this.params?.['source']?.[0] === 'map') {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  isCurrentlyNavigatedOn() {
+    const fullDetailPageTypes = ['area-restriction', 'bans-prohibitions', 'evac-orders', 'evac-alerts', 'danger-rating'];
+    if(fullDetailPageTypes.includes(this.params?.['type'])){
```
Comment (agordon-vivid) [inline client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]:
```suggestion
  isCurrentlyNavigatedOn(pageList) {

    if(pageList.includes(this.params?.['type'])){
```

**Review by agordon-vivid — CHANGES_REQUESTED** [review]
Updated to make more generic and clear what's being checked.

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts (hunk 4 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: Done

File: client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts (hunk 3 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts]: Done

**Pull request merged**